stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -610,24 +620,24 @@ end
 # Support `AbstractString`s, but avoid compilers needing to track backedges for callers
 # of these functions in case a user defines a new type that is `<: AbstractString`
 with_artifacts_directory(f::Function, artifacts_dir::AbstractString) =
-    with_artifacts_directory(f, string(artifacts_dir))
+    with_artifacts_directory(f, String(artifacts_dir)::String)
```

Comment:
These are just fixes so that the code adheres to what the comment claims it does.